### PR TITLE
[deepai] Add Types

### PR DESCRIPTION
**Pull request by NicolasNewman**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## deepai ([<kbd>unpkg</kbd>](https://unpkg.com/browse/deepai@latest/))
was missing the following properties:
1. axiosInstance
1. apiKey
1. renderResultIntoElement
1. renderAnnotatedResultIntoElement
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 61e430f60332909055c266194cadb81ffdb1b0ba
</p>

File: types/deepai/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Type definitions for deepai 1.0
+// Project: https://deepai.org/
+// Definitions by: NicolasNewman <https://github.com/NicolasNewman>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```
Comment (peterblazejewicz) [inline types/deepai/index.d.ts:4]:
```suggestion
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

export as namespace deepai;

```
IMO this is a script which supports global script and ships umd module:
https://github.com/deepai-org/deepai-js-client#installation
https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library

File: types/deepai/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    'summarization': {text: string};
+    'sentiment-analysis': {text: string};
+    'text-tagging': {text: string};
+}
+export interface ModelOutputs {
+    'torch-srgan': {id: string, output_url: string};
+    [key: string]: any;
+}
+export function setApiKey(apiKey: string): void;
+export function callStandardApi<T extends Models>(modelName: T, inputs_object: ModelInputs[T]): Promise<ModelOutputs[T]>;
```
Comment (peterblazejewicz) [inline types/deepai/index.d.ts:36]:
```suggestion
/** @async */
export function callStandardApi<T extends Models>(modelName: T, inputs_object: ModelInputs[T]): Promise<ModelOutputs[T]>;
```
https://github.com/deepai-org/deepai-js-client/blob/master/lib/core/DeepAI.js#L39

**Comment by peterblazejewicz** [comment]
what about those `render` methods, is that any useful without having them?

File: types/deepai/index.d.ts (hunk 1 above)
Comment (NicolasNewman) [reply types/deepai/index.d.ts:4]: You're right. In their webpack config they have the library type set to UMD.

**Comment by NicolasNewman** [comment]
> what about those `render` methods, is that any useful without having them?

The only two methods they have documented are `setApiKey()` and `callStandardApi()`. There's no reference to the `render()` methods anywhere else in their documentation or website so I'm unsure of what their purpose is or how they are meant to be used.

**Comment by peterblazejewicz** [comment]
ok, so you wrote this to cover your specific scenario when using 'torch-srgan' api, so when typed access to other kind of results is required, the extension of the typed result model is expected.
Is that right?

**Comment by NicolasNewman** [comment]
Correct. The creators of the API unfortunately didn't document what each endpoint returns and making calls isn't free. Since we're on an incredibly narrow budget, I was only able to provide explicit types for the endpoints we personally make use of. I designed the declaration file so if someone needs other endpoints defined they can easily submit a PR at a later time and add additional fields to `ModelOutputs`.

File: types/deepai/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for deepai 1.0
+// Project: https://deepai.org/
+// Definitions by: NicolasNewman <https://github.com/NicolasNewman>
```
Comment (peterblazejewicz) [inline types/deepai/index.d.ts:3]:
```suggestion
// Definitions by: Nicolas Newman <https://github.com/NicolasNewman>
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@NicolasNewman thanks!

**Comment by NicolasNewman** [comment]
Ready to merge

**Pull request merged**